### PR TITLE
Update create job process

### LINKED ISSUE #62: Implement new create job process
The process is mostly the same, but with a few added options in the configurations page (see mockups)

### PR DESCRIPTION
**Pull request by mas393**
This PR simplifies the create job process according to the mentor feedback. The changes implement our Figma mockups.

Changes include:
 - Hide advanced setting behind a dropdown
 - Have default "plastic" material option, and hide the more advanced material selection options
 - Add comment box, quantity, and color configuration options
 - Add upload card that is displayed when stl file is added to the dropbox, the file can be cancelled from the card

Closes #62

### CODE REVIEW

**Comment by mas393** [comment]
Screenshots of new configure page and file card for uploaded stl file:
Screenshot 2024-02-29 at 9 24 52 AM
Screenshot 2024-02-29 at 9 32 46 AM

**Comment by J-Raymer** [comment]
I think it might make things more simple if we move all other options other than plastic to the advanced settings. And have the help tip pointing the user to the advanced settings for pla, abs ect. We could also add other materials here such as metal ect, and if there are specific types they could be in advanced.

Also I think the default settings in advanced should be null by default. Not sure if that's how it is (reviewing on my phone)

For the complete by date maybe we could move that to advanced? So by default the customer is open to more printers. Maybe price is more important than completion time for example. Then the printers could list their expected completion time and cost in there bid

**Comment by mas393** [comment]
> I think it might make things more simple if we move all other options other than plastic to the advanced settings. And have the help tip pointing the user to the advanced settings for pla, abs ect. We could also add other materials here such as metal ect, and if there are specific types they could be in advanced.
> 

The other material options are hidden by default. The user has to hit a "More material options" button to expand into that view.

> Also I think the default settings in advanced should be null by default. Not sure if that's how it is (reviewing on my phone)
> 

I don't agree with this. Other printer service sites, as well as 3d slicer software provide sensible preset values.

> For the complete by date maybe we could move that to advanced? So by default the customer is open to more printers. Maybe price is more important than completion time for example. Then the printers could list their expected completion time and cost in there bid

Moving the completion date to the advanced tab is a good idea. But we are not adding any price details for milestone 2.

@J-Raymer

File: app/src/components/textForm.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-export default function TextForm({type, placeholder, onChange, value=""}) {
+export default function TextForm({type, placeholder, onChange, pattern, min, max, step, value=null}) {
```
Comment (Jasper-Nelligan) [inline app/src/components/textForm.js]: What's is pattern and step for? I don't see these parameters being used when creating a new textform

File: app/src/routes/configure.js [hunk 2]
```diff
@@ -1,72 +1,187 @@
...
+  const handleItemClick = (item) => {
+    setSelectedItem(item);
+    changeMaterial(item.Type)
+  };
+
+  return (
+    <div>
+      <button onClick={() => setShowOptions(true)}>
+        <div className="flex">
+          {showOptions ?
```
Comment (Jasper-Nelligan) [inline app/src/routes/configure.js:51]: I'm thinking this code would look cleaner if it was separated into a renderMaterials function, but that might just be my personal preference

**Review by Jasper-Nelligan — COMMENTED** [review]
Looks awesome! I love the addition of the new file card. One thing is that Complete By and Color don't keep their values when pressing back and then next again, but the rest of the fields do

**Review by Jasper-Nelligan — APPROVED** [review]

**Pull request merged**